store/sqlstore/post_store.go

```diff
@@ -2956,6 +2960,20 @@ func (s *SqlPostStore) savePostsPriority(transaction *sqlxTxWrapper, posts []*mo
 	return nil
 }
 
+func (s *SqlPostStore) savePostsPersistentNotifications(transaction *sqlxTxWrapper, posts []*model.Post) error {
+	for _, post := range posts {
+		if priority := post.GetPriority(); priority != nil && priority.PersistentNotifications != nil && *priority.PersistentNotifications {
```

Comment:
nit: We are looping here over multiple post and generating 2 queries per post, what is not specially but in the normal case (regular post creation), but can generate some problems in bulk creations (like imports).

I don't think is needed to fix that right away, but we can reduce the round trips to the database here following a different approach (for example getting the list of post with PersistentNotifications from a list of ids, that way you have the post that you need to create persistent notifications for. Probably there is even a way to do the insert based in a subselect (but that is going to make the query a bit more complicated and I don't think is going to improve that much the efficiency.

Reply:
It's 1 query(Insert) per post. `post.GetPriority()` gets data directly from the `post.Metadata` model.

Also, it's the same pattern similar to `updateThreadsFromPosts` and `savePostsPriority` in the very same function of saving post.